cli/command/node/cmd.go

```diff
@@ -38,6 +41,16 @@ func Reference(ctx context.Context, client apiclient.APIClient, ref string) (str
 		if err != nil {
 			return "", err
 		}
+		if info.Swarm.NodeID == "" {
+			// If there's no node ID in /info, the node probably
+			// isn't a manager. Call a swarm-specific endpoint to
+			// get a more specific error message.
+			_, err = client.NodeList(ctx, types.NodeListOptions{})
```

Comment:
My first approach was to use `info.Swarm.ControlAvailable`, but this misses important information that might cause alternate error messages, like the node being locked.

I think the best way is to either do what I'm doing here, or port this function to the client:

https://github.com/docker/docker/blob/250e05e42773a875d2fb8248b94fa72f2934a4b6/daemon/cluster/cluster.go#L313-L329